examples/next-graphql/components/helpers/ComponentRenderer.tsx

```diff
@@ -0,0 +1,34 @@
+import { Fragment } from 'react';
+import { VisualEditor } from '@/components/helpers/VisualEditor';
+
+interface ComponentsProps {
+  components: Array<JSX.Element>;
+  environment: string;
+}
+
+export function ComponentRenderer({ components, environment } : ComponentsProps) {
+  // If we are in preview mode, render the VisualEditor
+  if (environment === 'preview') {
+    const ids = components.map((component) => {
+      return component.props.paragraph.id;
+    });
+
+    const componentMapping: { [key: string]: JSX.Element } = components.reduce((acc, component) => {
```

Comment:
We don't need these type annotations
```suggestion
    const componentMapping = components.reduce((acc, component) => {
      acc[component.props.paragraph.id] = component;
      return acc;
    }, {} as { [key: string]: JSX.Element });
```

Reply:
let's blame copilot :P